cmd/update_test.go

```diff
@@ -0,0 +1,184 @@
+package cmd
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	cliVersion "github.com/oasisprotocol/cli/version"
+)
+
+func makeReleaseJSON(tag, body, assetName, assetURL string) string {
+	return fmt.Sprintf(`{
+	"tag_name": %q,
+	"body": %q,
+	"assets": [
+		{"name": %q, "browser_download_url": %q},
+		{"name": %q, "browser_download_url": %q}
+	]
+}`, tag, body, assetName, assetURL, assetName+".sha256", assetURL+".sha256")
+}
+
+func tarGz(t *testing.T, path, exeName string, content []byte) {
+	f, err := os.Create(path)
+	if err != nil {
+		t.Fatalf("create tarGz: %v", err)
+	}
+	gzw := gzip.NewWriter(f)
+	tw := tar.NewWriter(gzw)
+
+	hdr := &tar.Header{Name: exeName, Mode: 0o755, Size: int64(len(content))}
+	if err := tw.WriteHeader(hdr); err != nil {
+		t.Fatalf("hdr: %v", err)
+	}
+	if _, err := tw.Write(content); err != nil {
+		t.Fatalf("write: %v", err)
+	}
+	if err := tw.Close(); err != nil {
+		t.Fatalf("tw close: %v", err)
+	}
+	if err := gzw.Close(); err != nil {
+		t.Fatalf("gzw close: %v", err)
+	}
+	_ = f.Close()
+}
+
+func TestUpdate_PositiveFlow(t *testing.T) {
+	tmp := t.TempDir()
+	exeName := "oasis"
+	if runtime.GOOS == osWindows {
```

Comment:
Add windows to https://github.com/oasisprotocol/cli/blob/master/.github/workflows/ci-tests.yml#L22 so it's actually tested.